packages/teams-js/src/public/video.ts

```diff
@@ -154,7 +176,24 @@ export namespace video {
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
-    registerHandler('video.effectParameterChange', callback, false);
+
+    const effectParameterChangeHandler = (effectId: string | undefined): void => {
+      const promise = callback(effectId);
+      if (promise) {
+        promise
+          .then(() => {
+            sendMessageToParent('video.videoEffectReadiness', [true, effectId, EffectSuccessReason.EffectPrepared]);
+          })
+          .catch((reason) => {
+            // reason could be a string or an EffectFailureReason
```

Comment:
How do you know it's one of these two things? The returned promise is created by the application, they might be throwing a lot of different things out of here. You're going to need some sort of type checking to prevent invalid data potentially being sent up to the host.

Also it would be good if only one type of thing can be thrown: EffectFailureReason. Otherwise, you're passing an app-specified string up to the host, and we don't know what that might contain.

Reply:
updated, add type checking